src/server/services/csvPipeline/validateCsvUploadParams.js

```diff
@@ -0,0 +1,181 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const express = require('express');
+const { CSVPipelineError } = require('./CustomErrors');
+const { Param, EnumParam, BooleanParam, StringParam } = require('./ValidationSchemas');
+const failure = require('./failure');
+const validate = require('jsonschema').validate;
+
+// These are the default values of CSV Pipeline upload parameters.
+const DEFAULTS = {
+	common: {
+		gzip: 'true',
+		headerRow: 'false',
+		update: 'false'
+	},
+	meters: {
+	},
+	readings: {
+		createMeter: 'false',
+		cumulative: 'false',
+		cumulativeReset: 'false',
+		cumulativeResetStart: '0:00:00',
+		cumulativeResetEnd: '23:59:59',
```

Comment:
Thanks, done.

Reply:
As for the DB defaults of reading_length and reading_variation, I suppose these would correspond to the parameters length and lengthVariation. However, I have not change the default values for length and lengthVariation since the devDocs has specified certain defaults.

Reply:
I'm not so sure what to do about reaidng_length and reading_variation. Would they just have to appear in the meters csv file and so we do not have to specify them as parameters here?

Reply:
I agree that the meter will often have the values. Thinking about that, it would probably be best to load the meter values into the form but I think that is something for the future given it would be an update once the meter name is known. Given the DB will either have a set value or the default value, we should let it be used unless a specific value is given in the form/curl request. Thus, I have changed my mind and now think we should not have default values in the form for these values so the pipeline will use the DB defaults unless a new value is given. Thus, src/client/app/utils/csvUploadDefaults.ts would not set a default for cumulative for those 4 values. I think the other defaults are fine. Unfortunately we don't have duplication in the DB but we may change how it works so that is probably okay at this point (making default 1). Does this seem okay as a change?

Reply:
This change sounds okay. To clarify, does this mean that we won't have the cumulative values in the request anymore as it would be known in the database?

Reply:
Sorry for the confusion. What I think might be best is to allow the user to input a value (to override what is in the DB) but not to give default values in the form. Thus, I expect the pipeline to use the DB  values if nothing is given in the form but use any provided values as a override. Another consequence of this change is that the form and curl requests will act similarly for these values.

Reply:
Thanks for clarifying. This is interesting because that means that the cumulative options are optional. Now that I think about it, checkboxes may give the illusion that not ticking means that option is ignored, but they are implicitly saying `false`. I will think more on whether or not I should use other button types. Maybe a radio button for true/false or a select button to _really_ ignore the option and rely on server/db defaults.

Reply:
This is interesting because it means that the cumulative checkboxes need to represent three values (true,false,meter value) rather than just two values (true,false). I might have to change the button type to more appropriately serve this purpose. I will think more about it.

Reply:
I decided not to change the check boxes. It would easier to just tick the check boxes depending on the value of the meter served from the DB, which would be a future task. The default value is undefined for parameters whose default should be read from the DB. I elaborate in `/src/server/services/csvPipeline/validateCsvUploadParams.js` in the recent commit 22a2252. Otherwise, done.